packages/command/src/sfdxCommand.ts

```diff
@@ -234,7 +234,7 @@ export abstract class SfdxCommand extends Command {
       }
     } catch (err) {
       if (this.statics.requiresUsername) {
-        if (err.name === 'NoUsername') {
+        if (err.name === 'NoUsername' || err.name === 'AuthInfoCreationError') {
           throw SfdxError.create('@salesforce/command', 'command', 'RequiresUsernameError');
```

Comment:
I kinda liked the messaging you had. Should we update the message or is it fine how it is?